criu/util.c

```diff
@@ -1880,11 +1880,13 @@ uint64_t criu_run_id;
 
 void util_init(void)
 {
-	struct timespec tp;
+	struct stat statbuf;
 
-	clock_gettime(CLOCK_MONOTONIC, &tp);
-	criu_run_id = ((uint64_t)getpid() << 32) + tp.tv_sec + tp.tv_nsec;
+	criu_run_id = getpid();
+	if (!stat("/proc/self/ns/pid", &statbuf))
+		criu_run_id |= (uint64_t)statbuf.st_ino << 32;
```

Comment:
fall back to the clock_gettime way if /proc/self/ns/pid isn't accessible.  Conflicts with clock_gettime() are almost impossible.

Reply:
If pidns is not enabled in kernel, then names based on CRIU's pid alone are already machine-level unique.

Reply:
In this case, you need to check that errno is ENOENT and report errors in other cases.